src/connector/with_options_source.yaml

```diff
@@ -455,9 +455,31 @@ PulsarProperties:
   - name: oauth.scope
     field_type: String
     required: false
-  - name: s3_credentials
-    field_type: HashMap<String,String>
-    comments: required keys refer to [`crate::aws_utils::AWS_DEFAULT_CONFIG`]
+  - name: region
+    field_type: String
+    required: false
+  - name: endpoint
+    field_type: String
+    required: false
+    alias: endpoint_url
+  - name: access_key
+    field_type: String
+    required: false
+  - name: secret_key
+    field_type: String
+    required: false
+  - name: session_token
+    field_type: String
+    required: false
+  - name: arn
+    field_type: String
+    required: false
+  - name: external_id
+    field_type: String
+    comments: This field was added for kinesis. Not sure if it's useful for other connectors.  Please ignore it in the documentation for now.
```

Comment:
`external_id` serves as a common part for aws auth, not only related to kinesis but also s3. 
I am also drafting a doc for aws auth usage, demonstrating how these fields work together.